test/java/seedu/addressbook/storage/StorageFileTest.java

```diff
@@ -0,0 +1,109 @@
+package seedu.addressbook.storage;
+
+import static org.junit.Assert.assertEquals;
+import java.nio.file.Paths;
+import java.util.Collections;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.Address;
+import seedu.addressbook.data.person.Email;
+import seedu.addressbook.data.person.Name;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.tag.Tag;
+import seedu.addressbook.data.tag.UniqueTagList;
+import seedu.addressbook.storage.StorageFile.StorageOperationException;
+import static seedu.addressbook.util.TestUtil.assertTextFilesEqual;
+
+public class StorageFileTest {
+    private static final String TEST_DATA_FOLDER = "test/data/StorageFileTest";
+    private static final String TEST_TEMP = "temp.txt";
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Rule
+    public TemporaryFolder testFolder = new TemporaryFolder();
+
+    @Test
+    public void constructor_nullFilePath_exceptionThrown() throws Exception {
+        thrown.expect(NullPointerException.class);
+        new StorageFile(null);
+    }
+
+    @Test
+    public void constructor_noTxtExtension_exceptionThrown() throws Exception {
+        thrown.expect(IllegalValueException.class);
+        new StorageFile(TEST_DATA_FOLDER + "/InvalidfileName");
+    }
+
+    @Test
+    public void load_invalidFormat_exceptionThrown() throws Exception {
+        // The file contains valid xml data, but does not match the AddressBook class
+        StorageFile storage = getStorage("/InvalidData.txt");
+        thrown.expect(StorageOperationException.class);
+        storage.load();
+    }
+
+    @Test
+    public void load_validFormat() throws Exception {
+        AddressBook actualAB = getStorage("/ValidData.txt").load();
+        AddressBook expectedAB = getTestAddressBook();
+
+        // ensure loaded AddressBook is properly constructed with test data
+        // TODO: overwrite equals method in AddressBook class and replace with equals method below
+        assertEquals(actualAB.getAllPersons(), expectedAB.getAllPersons());
+    }
+
+    @Test
+    public void save_nullAddressBook_exceptionThrown() throws Exception {
+        StorageFile storage = getTempStorage();
+        thrown.expect(NullPointerException.class);
+        storage.save(null);
+    }
+
+    @Test
+    public void save_validAddressBook() throws Exception {
+        AddressBook ab = getTestAddressBook();
+
+        getTempStorage().save(ab);
+
+        // ensure xml data for sample data is properly structured and saved
+        assertSaveSuccess("ValidData.txt");
```

Comment:
Method name doesn't seem to describe what we want to test for very well.

All we want to do is to compare the contents of two files, it is just that we can't easily get the paths of the files we want to compare (since we don't have suitable methods for them).

One way out of this is to:

Introduce a method `assertStorageFilesEqual(StorageFile, StorageFile)`:
```java
    private void assertStorageFilesEqual(StorageFile a, StorageFile b) throws Exception {
        assertTextFilesEqual(Paths.get(a.getPath()), Paths.get(b.getPath()));
    }
```

We can then replace the above call to `assertSaveSuccess()` with:
```java
assertStorageFilesEqual(tempStorage, getStorage("ValidData.txt"));
```

Which is much less opaque.